org.eclipse.sisu.plexus/src/main/java/org/eclipse/sisu/plexus/CompositeBeanHelper.java

```diff
@@ -264,21 +264,33 @@ private Object convertProperty( final Class<?> beanType, final Class<?> rawPrope
                                             listener );
     }
 
-    private static Method findMethod( final Class<?> beanType, final Type[] paramTypeHolder, final String methodName )
-    {
+    private Method findMethod( final Class<?> beanType, final Type[] paramTypeHolder, final String methodName, final Class<?> valueType ) {
+        Method candidate = null;
         for ( final Method m : beanType.getMethods() )
         {
             if ( methodName.equals( m.getName() ) && !Modifier.isStatic( m.getModifiers() ) )
             {
                 final Type[] paramTypes = m.getGenericParameterTypes();
                 if ( paramTypes.length == 1 )
                 {
+                    if ( valueType != null )
+                    {
+                        if(m.getParameters()[0].getType().isAssignableFrom(valueType))
+                        {
+                            paramTypeHolder[0] = paramTypes[0];
+                            return m;
+                        }
+                    }
                     paramTypeHolder[0] = paramTypes[0];
```

Comment:
I think this assignment should be moved inside the `if(candidate == null)` block below, otherwise the returned param type won't be the type captured for the candidate but would be the type from the last method in the loop...

Reply:
@mcculls Thanks for the review! change made with https://github.com/eclipse/sisu.plexus/pull/52/commits/7e4271488804fca571c8d2706ed19c6be2fcee3e